pkg/manifest/ostree_deployment.go

```diff
@@ -353,6 +355,17 @@ func (p *OSTreeDeployment) serialize() osbuild.Pipeline {
 			},
 		}))
 
+		// This will create a custom systemd unit that create
+		// mountpoints if its not present.This will safeguard
+		// any ostree deployment  which has custom filesystem
+		// during ostree upgrade.
+		// issue # https://github.com/osbuild/images/issues/352
+		if len(p.CustomFileSystems) != 0 {
+			serviceName := "osbuild-ostree-mountpoints.service"
+			pipeline.AddStage(osbuild.NewSystemdUnitCreateStageOptions(createMountpointService(serviceName, p.CustomFileSystems)))
```

Comment:
You need to call
```
<stage_object>.MountOSTree(p.osName, ref, 0)
```
before adding it to the pipeline. Have a look at other lines around your added code.

Reply:
updated